packages/@pollyjs/adapter/src/index.js

```diff
@@ -52,36 +58,6 @@ export default class Adapter {
     }
   }
 
-  shouldReRecord(pollyRequest, recordingEntry) {
-    const { config } = pollyRequest;
-
-    if (isExpired(recordingEntry.startedDateTime, config.expiresIn)) {
-      if (!config.recordIfExpired) {
-        console.warn(
-          '[Polly] Recording for the following request has expired but `recordIfExpired` is `false`.\n' +
-            `${recordingEntry.request.method} ${recordingEntry.request.url}\n`,
-          recordingEntry
-        );
-
-        return false;
-      }
-
-      if ('navigator' in global && !navigator.onLine) {
-        console.warn(
-          '[Polly] Recording for the following request has expired but the browser is offline.\n' +
-            `${recordingEntry.request.method} ${recordingEntry.request.url}\n`,
-          recordingEntry
-        );
```

Comment:
Note: we removed this part of the logic in this PR. If we were to include it, we would only need to include it for `expiryStrategy: 'record'`, and we think that it makes more sense to move between expiryStrategies (from record to warn, for example) for this offline case. If there's a case we're not thinking of, let us know and we'll update it. The branching got pretty hard to understand if this were just nested as-is in the `record` case.

Reply:
We're also unsure why this logic would only apply to the expired case and not other "background record" cases, such as `recordIfMissing`.

Reply:
Just to clarify, we're referencing the `navigator.onLine` behavior, not recording in general :)

Reply:
> We're also unsure why this logic would only apply to the expired case and not other "background record" cases, such as recordIfMissing.

I totally agree with you here. I think instead of removing this, it should be moved directly into the `record` method.